versioned/storage/cassandra/src/main/java/org/projectnessie/versioned/storage/cassandra/CassandraBackend.java

```diff
@@ -329,30 +333,38 @@ boolean executeCas(BoundStatement stmt) {
   }
 
   ResultSet execute(BoundStatement stmt) {
-    return session.execute(stmt);
+    try {
+      return session.execute(stmt);
+    } catch (DriverException e) {
+      throw unhandledException(e);
+    }
   }
 
   CompletionStage<AsyncResultSet> executeAsync(BoundStatement stmt) {
     return session.executeAsync(stmt);
   }
 
-  void handleDriverException(DriverException e) {
-    if (e instanceof CASWriteUnknownException) {
-      logCASWriteUnknown((CASWriteUnknownException) e);
+  static RuntimeException unhandledException(DriverException e) {
+    if (e instanceof QueryExecutionException) {
+      return new UnknownOperationResultException(e);
+    } else if (e instanceof DriverTimeoutException
+        || e instanceof NodeUnavailableException
+        || e instanceof AllNodesFailedException) {
+      return new UnknownOperationResultException(e);
     } else {
-      throw e;
+      return e;
     }
   }
 
-  @SuppressWarnings("Slf4jDoNotLogMessageOfExceptionExplicitly")
-  private static void logCASWriteUnknown(CASWriteUnknownException e) {
-    if (LOGGER.isDebugEnabled()) {
-      LOGGER.debug(
-          "Assuming CAS failed due to CASWriteUnknownException with message'{}', coordinator: {}, errors: {}, warnings: {}",
-          e.getMessage(),
-          e.getExecutionInfo().getCoordinator(),
-          e.getExecutionInfo().getErrors(),
-          e.getExecutionInfo().getWarnings());
+  static void handleDriverException(DriverException e) {
+    if (e instanceof QueryExecutionException) {
+      throw new UnknownOperationResultException(e);
+    } else if (e instanceof DriverTimeoutException
+        || e instanceof NodeUnavailableException
+        || e instanceof AllNodesFailedException) {
+      throw new UnknownOperationResultException(e);
```

Comment:
Isn't it the same as the first `if`? Why have a separate `if` block?

Reply:
Sorry, I do not see any similarity.